File: result/translate/chatgpt/java2cpp/c++/FREQUENT_ELEMENT_ARRAY_1.cpp
int f_gold(int arr[], int n) {
    unordered_map<int, int> hp;
    for (int i = 0; i < n; i++) {
        int key = arr[i];
        if (hp.find(key) != hp.end()) {
            int freq = hp[key];
            freq++;
            hp[key] = freq;
        } else {
            hp[key] = 1;
        }
    }
    int max_count = 0, res = -1;
    for (auto val : hp) {
        if (max_count < val.second) {
            res = val.first;
            max_count = val.second;
        }
    }
    return res;
}

File: result/translate/chatgpt/java2python/python/FREQUENT_ELEMENT_ARRAY_1.py
def f_gold(arr, n):
    hp = {}
    
    for i in range(n):
        key = arr[i]
        if key in hp:
            freq = hp[key]
            freq += 1
            hp[key] = freq
        else:
            hp[key] = 1
    
    max_count = 0
    res = -1
    
    for key, value in hp.items():
        if max_count < value:
            res = key
            max_count = value
    
    return res